### DISCUSSION TITLE
Menus, external stylesheet and hot reloading

### DISCUSSION THREAD

**Discussion by rohitsathish** [zauberzeug/nicegui#1834] [Q&A] [answered]
### Question

1. I can't seem to get the menu to open. The image appears and so does the click animation, but the menu doesn't open.
2. Is there any way to use Tailwind (or Quasar classes) with an external stylesheet? I prefer to have all my styling on a separate sheet.
3. Is there any way to get ```uvicorn_reload_includes``` to work with detecting changes in .css files? ```uvicorn_reload_includes="*.css"``` doesn't seem to work.

Any help is appreciated, thanks.

Reference code
```py
# app.py

from pathlib import Path
from nicegui import ui

@ui.page("/")
def main_page():
    ui.add_head_html(
        f"<style>{(Path(__file__).parent / 'style.css').read_text()}</style>"
    )

    with ui.row().classes("w-full items-center justify-between custom-navbar"):
        # Left-aligned icon and title
        with ui.row().classes("items-center"):
            ui.icon("home").classes("text-2xl custom-icon")
            ui.label("My App").classes("custom-title")

        # Center-aligned links
        with ui.row().classes("items-center justify-center flex-grow"):
            ui.link("Home", "/home").classes("custom-link")
            ui.link("Search", "/search").classes("custom-link")
            ui.link("About", "/about").classes("custom-link")

        # Right-aligned settings dropdown
        with ui.row().classes("items-center justify-end"):
            with ui.button(icon="menu").classes("custom-menu-button"):
                with ui.menu() as menu:
                    ui.menu_item("Close", on_click=menu.close)

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
1. I can't reproduce a problem with the menu. It opens as it should.
2. As far as I know there is no way to use _classes_ in an external stylesheet. I recommend something like a style.py with custom Python classes
    ```py
    class custom_menu_button(ui.button):
        def __init__(self):
            super().__init__(icon='menu')
            self.classes("custom-menu-button")
    ```
    or functions
    ```py
    def custom_menu_button():
        return ui.button(icon='menu').classes("custom-menu-button")
    ```
3. Auto-reload should also work for CSS files. I usually run my apps with `uvicorn_reload_includes='*.py, *.html, *.js, *.css'`.

**Comment by ed2050** [comment]
@rohitsathish @falkoschindler 
I had same problem, but fixed it.  Couldn't get uvicorn_reload_includes to work.  Finally fixed it by completely stopping the script and relaunching it.  Change detection doesn't seem to pickup changes to ui.run args.  Maybe this should be mentioned on ui.run page.

FYI the [docs page](https://nicegui.io/documentation/run) is incorrect.  It says default glob pattern is `'.py'`.  That's not a valid glob pattern.  Technically it is valid, but has no wildcards.  Only matches the literal string.  

Docs should say `*.py` instead.  This was confusing as I thought it might be part of the problem.

**Reply by rodja** [reply]
Thanks for bringing this up. I've pushed two tiny changes to main.
